src/middlewared/middlewared/plugins/webdav.py

```diff
@@ -237,6 +237,10 @@ async def pool_post_import(middleware, pool):
     """
     Makes sure to reload WebDAV if a pool is imported and there are shares configured for it.
     """
+    if pool is None:
+        asyncio.ensure_future(middleware.call('etc.generate', 'webdav'))
+        return
+
     path = f'/mnt/{pool["name"]}'
     if await middleware.call('sharing.webdav.query', [
```

Comment:
what if we combine both statements?

if pool is None or ...:
    asyncio.ensure_future(middleware.call('service.reload', 'webdav'))

Reply:
My thought was to avoid a restart of apache in this case because the service's rc script will take care of starting it later in boot.

Reply:
Ah, makes sense!